ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -154,8 +155,40 @@ public KsqlResource(
     this.registerKsqlStatementTasks();
   }
 
+  @POST
+  @Path("/terminate")
+  public Response terminateCluster(final KsqlRequest request) {
```

Comment:
I still think it should be this handler that deals with deleting topics. The terminate written to the command topic should only tell nodes to stop handling requests and stop all queries.